internal/packager/fork.go

```diff
@@ -34,44 +34,34 @@ func Fork(originName, forkName, outputDir string, maintainers []string) error {
 	namespace, name := splitPackageName(forkName)
 	appDir := filepath.Join(outputDir, internal.DirNameFromAppName(name))
 	if err := os.MkdirAll(appDir, 0755); err != nil {
-		return err
+		return errors.Wrap(err, "failed to create output application directory")
 	}
 
-	// iterate on contents
-	for k, vs := range payload {
-		v := []byte(vs)
-		// Deal with windows/linux slashes
-		convertedFilepath := filepath.FromSlash(k)
-
-		// Check we aren't doing ./../../../ etc in the path
-		fullFilepath := filepath.Join(appDir, convertedFilepath)
-		_, err := filepath.Rel(appDir, fullFilepath)
-		if err != nil {
-			log.Warnf("dropping image entry '%s' with unexpected path outside of app dir", k)
-			continue
-		}
-		if k == internal.MetadataFileName {
-			log.Debug("Loading app metadata")
-			v, err = updateMetadata(v, namespace, name, maintainers)
-			if err != nil {
-				return err
-			}
-		}
+	// Create all the files on disk
+	if err := ExtractImagePayloadToDiskFiles(appDir, payload); err != nil {
+		return err
+	}
 
-		// Create the directories for any nested files
-		basepath := filepath.Dir(fullFilepath)
-		os.MkdirAll(basepath, os.ModePerm)
-		if err := ioutil.WriteFile(fullFilepath, v, 0644); err != nil {
-			return errors.Wrapf(err, "failed to write output file: %s", fullFilepath)
-		}
+	// Update the metadata file
+	fullFilepath := filepath.Join(appDir, internal.MetadataFileName)
+	bytes, err := ioutil.ReadFile(fullFilepath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read metadata file from: %s", fullFilepath)
+	}
+	log.Debug("Loading app metadata")
```

Comment:
I'm not sure why this line is here - it was there when I arrived, let me know if I should delete it :)

Reply:
I don't think that we need it. @shin- worked on this originally and might have more context though.

Reply:
Have removed it for now. Will add back in needed